### PR TITLE
feat: ContextModel

### LINKED ISSUE #1885: sequelize v3 支持 ctx 的模式
fork 了一个版本，基于 v3 实现 ctx class 模式，方便让 sql 请求与 http ctx 上下文发生关联。

https://github.com/eggjs/sequelize/tree/v3

## 需要实现的功能

- [ ] 基于 ctx 的 model
- [ ] mysql 使用 mysql 模块自身的 pool 实现
- [ ] 提供 before query 事件，让日志打印可以个性化

### PR DESCRIPTION
**Pull request by fengmk2**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

Closes https://github.com/eggjs/egg/issues/1885

### CODE REVIEW

**Comment by fengmk2** [comment]
还差 associations 的修改。

File: lib/model.js [hunk 1]
```diff
@@ -1240,6 +1241,57 @@ class Model {
...
+      // Contextify all assocation Models first
+      const association = Object.create(ContextModel.associations[associationAccessor]);
+      if (association.source) {
+        association.source = ContextModel;
+      }
+      if (association.target) {
+        association.target = association.target.contextify(ctx);
+      }
+
+      association.injectAttributes();
```
Comment (fengmk2) [inline lib/model.js]: @huacnlee v4 简单了许多，都统一为 injectAttributes 和 association.mixin 了，不需要区分 association 的类型了。

File: lib/model.js [hunk 2]
```diff
@@ -1240,6 +1241,62 @@ class Model {
...
+      const association = Object.create(ContextModel.associations[associationAccessor]);
+      if (association.source) {
+        association.source = association.source.contextify(ctx);
+      }
+      if (association.target) {
+        association.target = association.target.contextify(ctx);
+      }
+
+      // can skip injectAttributes on association
+      // association.injectAttributes();
```
Comment (fengmk2) [inline lib/model.js:1284]: injectAttributes 在 v4 版本里面也能够跳过，它没有任何上下文绑定。

**Comment by fengmk2** [comment]
后面交给 @huacnlee

**Comment by huacnlee** [comment]
https://github.com/eggjs/sequelize/pull/1/commits/b95fc3701b371912af88b13415d6371e4291cd87

修了一下测试，同样的 Model, 同样的 ctx, `contextify` 以后应该是相同的结果。

**Pull request closed without merging**